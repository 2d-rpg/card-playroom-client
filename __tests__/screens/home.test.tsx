import React from "react";
import { render, fireEvent, waitFor } from "@testing-library/react-native";

import HomeScreen from "../../src/screens/home/index";

describe("<HomeScreen />", () => {
  // snapshot
  it("should match snapshot", () => {
    const rendered = render(<HomeScreen />);
    expect(rendered.toJSON()).toMatchSnapshot();
  });

  // default text
  it("should match default value", () => {
    const { getByTestId } = render(<HomeScreen />);
    expect(getByTestId("RNE__Input__text-input").props.value).toEqual(
      "127.0.0.1"
    );
  });
});
