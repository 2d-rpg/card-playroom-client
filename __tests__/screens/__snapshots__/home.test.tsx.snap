// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen /> should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "paddingHorizontal": 10,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#86939e",
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      サーバーアドレス
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "borderColor": "#86939e",
          "flexDirection": "row",
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        editable={true}
        onChangeText={[Function]}
        placeholderTextColor="#86939e"
        rejectResponderTermination={true}
        style={
          Object {
            "alignSelf": "center",
            "borderColor": "#ccc",
            "borderWidth": 1,
            "color": "#242424",
            "flex": 1,
            "fontSize": 18,
            "minHeight": 40,
          }
        }
        testID="RNE__Input__text-input"
        underlineColorAndroid="transparent"
        value="127.0.0.1"
      />
    </View>
    <Text
      style={
        Object {
          "color": "#ff190c",
          "fontSize": 12,
          "margin": 5,
        }
      }
    />
  </View>
</View>
`;
